internal/venice-client-common/src/main/java/com/linkedin/venice/compression/ZstdWithDictCompressor.java

```diff
@@ -37,38 +45,100 @@ public byte[] compress(byte[] data) {
   }
 
   @Override
-  public ByteBuffer compress(ByteBuffer data) {
+  public ByteBuffer compress(ByteBuffer data, int startPositionOfOutput) throws IOException {
+    long maxDstSize = Zstd.compressBound(data.remaining());
+    if (maxDstSize > Integer.MAX_VALUE) {
+      throw new ZstdException(Zstd.errGeneric(), "Max output size is greater than Integer.MAX_VALUE");
+    }
+    int sizeOfOutput = (int) maxDstSize + startPositionOfOutput;
```

Comment:
Is it possible that `sizeOfOutput` will be greater than `Integer.MAX_VALUE`?
It seems safer to update the check to be against `sizeOfOutput` instead.

Reply:
So essentially, you're saying we should throw `if ((maxDstSize + startPositionOfOutput) > Integer.MAX_VALUE)` ?

Reply:
Yes

Reply:
Done